lib/serial_terminal.pm

```diff
@@ -55,6 +89,25 @@ sub login {
     assert_script_run('echo Logged into $(tty)', $bmwqemu::default_timeout, result_title => 'vconsole_login');
 }
 
+=head2 select_virtio_console
+
+   select_virtio_console();
+
+Helper to select console 'root-virtio-terminal' or 'root-console'.
+=cut
+# TODO: Move here optional init with add_serial_console($console);
+sub select_virtio_console {
+    my $is_zkvm = get_var('S390_ZKVM');
+    if (get_var('VIRTIO_CONSOLE') and !$is_zkvm) {
+        select_console('root-virtio-terminal');
+        return 1;
+    }
+
+    bmwqemu::fctwarn("Cannot use virtio console on s390 on ZKVM") if $is_zkvm;
```

Comment:
We probably only want to print a warning if VIRTIO_CONSOLE is set because someone can safely call this function when it is not set.

Also `&&` is preferred over `and`